pkg/server/server_test.go

```diff
@@ -607,6 +607,7 @@ func TestListObjectsReleasesConnections(t *testing.T) {
 	s := MustNewServerWithOpts(
 		WithDatastore(storagewrappers.NewContextWrapper(ds)),
 		WithMaxConcurrentReadsForListObjects(1),
```

Comment:
@jon-whit is `WithMaxConcurrentReadsForListObjects(1),` relevant to this test? I think not 🤔 

If it is relevant, do we need to add `WithMaxConcurrentReadsForListUsers(1)` to make this new test useful?

If it is not relevant, let's remove it @willvedd.

Reply:
It appears as though it is - we use it to give better confidence that we have blocked in flight Goroutines when the context is cancelled.

Reply:
@miparnisari @jon-whit The tests will pass with or without that config and if its configured with any non-zero positive number. This suggests that `WithMaxConcurrentReadsForListObjects(1),` is not essential in this test. 

I'm actually not even able to make this test fail with any change in ListUsers. Is this test even necessary?

Reply:
Yes, i'd keep it. It may not be applicable to the current implementation, but it's useful to know that we are not hogging connections after the API call ends.